### DISCUSSION TITLE
Dataset with string

### DISCUSSION THREAD

**Discussion by manonromain** [apache/incubator-mxnet#15359] [Q&A]
I want to train a CV network with MXNet and I created a custom Dataset class (inherited from SegmentationDataset in GluonCV - works well) but I'd now like this class to return the name of the image file (a string).

Problem is MXNet refuses to batch strings.

So I tried to create a NDArray from my string but NDArray won't take strings as dtype. What should I do?

**Comment by vdantu** [comment]
@mxnet-label-bot add [question, gluon]
